classes/local/manager.php

```diff
@@ -135,6 +135,7 @@ public static function update_object_by_hash($contenthash, $newlocation) {
         global $DB;
         $newobject = new stdClass();
         $newobject->contenthash = $contenthash;
+        $newobject->filesize = $DB->get_field('files', 'filesize', ['contenthash' => $contenthash], IGNORE_MULTIPLE);
```

Comment:
Do we strictly need this here? In theory it should already have a size. The adhoc task should be back filling this data. If we want to deal with the cases where we are touching this table before the adhoc task has completed then lets make it only do this if the filesize is missing.

Reply:
Also in some cases the calling code already has the filesize and could optionally pass it in save another db read